### DISCUSSION TITLE
Sass vs template literals

### DISCUSSION THREAD

**Discussion by blackfalcon** [AlaskaAirlines/WC-Generator#436] [Announcements]
## Is your feature request related to a problem? Please describe.

Processing Sass to CSS takes time...

## Describe the solution you'd like

What about replacing Sass with JS/CSS like so;

https://twitter.com/daKmoR/status/1191347381700321281 or https://vanilla-extract.style/documentation/

What are our opportunities? What does this get us? What will this cost us? 

## Describe alternatives you've considered

- https://github.com/AlaskaAirlines/WC-Generator/issues/422

**Comment by geoffrich** [comment]
I'm not super familiar with the suggested libraries, but I want to make sure we take into account that many teams are currently importing [various Auro Sass utility files](https://github.com/search?q=org%3AAlaska-ECommerce+webcorestylesheets&type=code). Any switch needs to take those dependencies into account as well and consider what a migration strategy for other teams would look like. What works for how we consume styles in a web component may not work for how teams consume styles in a React/Svelte/other app.

I don't know if the scope of this issue is meant to encompass WCSS as well as the Sass used inside components, but wanted to make sure these considerations were included in the discussion. I'm not opposed to moving away from Sass, but it's tricky when other team's build processes are already set up to consume Sass resources and may require re-work to consume a new format.

If component build time/DX is one of the main reasons, there may be opportunities to speed up the current pipeline, especially as the need to generate IE fallbacks goes away.

**Comment by blackfalcon** [comment]
@geoffrich the scope of this issue is to consider not using Sass for HTML custom elements only. This has no impact on anyone using WCSS or the use of tokens. 

This issue is to discuss what a world looks like for doing all the local element's CSS with only CSS and support from JS/CSS.
